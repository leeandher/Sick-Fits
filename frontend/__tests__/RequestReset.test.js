import { mount } from 'enzyme'
import wait from 'waait'
import toJSON from 'enzyme-to-json'
import { MockedProvider } from 'react-apollo/test-utils'

import RequestReset, {
  RESET_REQUEST_MUTATION,
} from '../components/RequestReset'

const mocks = [
  {
    request: {
      query: RESET_REQUEST_MUTATION,
      variables: { email: 'me@leander.xyz' },
    },
    result: {
      data: { requestReset: { message: 'success', __typename: 'Message' } },
    },
  },
]

describe('<RequestReset />', () => {
  it('renders and matches the snapshot', async () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>,
    )
    await wait()
    wrapper.update()
    const form = wrapper.find('form[data-test="form"]')
    expect(toJSON(form)).toMatchSnapshot()
  })
  it('calls the mutation', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <RequestReset />
      </MockedProvider>,
    )
    // Simulate typing an email
    wrapper.find('input').simulate('change', {
      target: { name: 'email', value: 'me@leander.xyz' },
    })
    wrapper.find('form[data-test="form"]').simulate('submit')
    // Submit the form
    await wait()
    wrapper.update()
    expect(wrapper.debug()).toContain('success')
  })
})
