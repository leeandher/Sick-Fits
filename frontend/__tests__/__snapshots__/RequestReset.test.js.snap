// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset /> renders and matches the snapshot 1`] = `
<form
  className="Form-sc-1xszr8q-0 ZWDQB"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Forgot your password?
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="email"
    >
      Email
      <input
        name="email"
        onChange={[Function]}
        placeholder="e.g. kanye@west.com"
        type="text"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Request Reset!
    </button>
  </fieldset>
</form>
`;
